primary/audio_hw.c

```diff
@@ -1353,7 +1356,10 @@ static int adev_open_input_stream(struct audio_hw_device *dev,
             }
         }
     }
-
+    if(!params) {
+        adev->cardc = get_pcm_card("Dummy");
+        params = pcm_params_get(adev->cardc, PCM_DEVICE, PCM_IN);
```

Comment:
Can we handle error situation like this         if (!params)            return -ENOSYS; as before ?